File: otani/php/09.php
<?php
$str = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind .";
$str = divide_word($str);

$str_partial = array();
foreach($str as $key => $word){
  if(strlen($word) >= 5){
    $str_partial[$key] = substr($word, 1, strlen($word) - 2);
    $str_partial[$key] = str_shuffle($str_partial[$key]);
    $str[$key] = $str[$key][0]. $str_partial[$key]. $str[$key][strlen($word) - 1];
  }else{
    $str_partial[$key] = $word;
  }
}
var_dump($str);

function divide_word($str){
  $str_rep1 = str_replace(",", "", $str);
  $str_rep2 = str_replace(".", "", $str_rep1);
  $str_rep3 = str_replace(":", "", $str_rep2);
  $str_rep4 = str_replace("  ", " ", $str_rep3);
  $str_exp = explode(" ", $str_rep4);
  return $str_exp;
}

?>

File: otani/python/1/09.py
import re
import random
from pprint import pprint

string = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."
string = string.replace(",", "")
string = string.replace(".", "")
string = string.split(" ")

typoglycemia = []
for word in string:
    if(len(word) > 4):
        random_word = list(word[1:len(word) - 1])
        random.shuffle(random_word)
        random_word = ''.join(random_word)
        word = word[0:1] + random_word + word[-1]
    typoglycemia.append(word)
pprint(typoglycemia)
